packages/pangea-sdk/pangea/asyncio/services/file_scan.py

```diff
@@ -95,9 +106,9 @@ async def file_scan(
                 size = params.size
             else:
                 crc, sha, size = None, None, None
-            files: List[Tuple] = [("upload", ("filename", file, "application/octet-stream"))]
-        else:
-            raise ValueError("Need to set file_path or file arguments")
+            files = [("upload", ("filename", file, "application/octet-stream"))]
+        elif source_url is None:
+            raise ValueError("Need to set one of `file_path`, `file`, or `source_url` arguments.")
```

Comment:
Should we check that when setting `file` or `file_path`, `source_url` is empty?

Reply:
We could, but what I'm trying to do here is lower the delta between `beta` and `main`, so let's do that in a separate pair of PRs.